syntax = "proto3"; //版本

package fixbug; //声明包

//生成service服务类和rpc函数
option cc_generic_services = true;

message ResultCode
{
    int32 errcode = 1;
    bytes errmsg = 2;
}

//定义登录请求消息类型
message LoginRequest
{
    bytes name = 1;
    bytes pwd = 2;
}

//定义登陆响应消息类型
message LoginResponse
{
    ResultCode result = 1;
    bool success = 2;
    
}

message GetFriendListRequest
{
    uint32 userid = 1;
}

message User
{
    bytes name = 1;
    uint32 age = 2;
    enum Sex
    {
        MAN = 0;
        WOMAN = 1;
    }
    Sex sex = 3;
}

message GetFriendListResponse
{
    ResultCode result = 1;
    //列表类型 repeated
    repeated User friend_list = 2;
}

//定义rpc函数类型 - service
service UserServiceRpc
{
    rpc login(LoginRequest) returns(LoginResponse);
}

